ufm-test-services/metamask/tests/prometheusUtils.ts

```diff
@@ -0,0 +1,70 @@
+import 'dotenv/config'
+import fetch from 'node-fetch'
+import { Counter, Pushgateway } from 'prom-client'
+
+const txSuccessMetricName = 'metamask_tx_success'
+const txFailureMetricName = 'metamask_tx_failure'
+
+const txSuccessCounter = new Counter({
+  name: txSuccessMetricName,
+  help: 'A counter signifying the number of successful transactions sent with Metamask since last failure',
+})
+const txFailureCounter = new Counter({
+  name: txFailureMetricName,
+  help: 'A counter signifying the number of failed transactions sent with Metamask since last successful transaction',
+})
+
+export const getMetricValueFromPrometheus = async (metricName: string) => {
+  if (process.env.PROMETHEUS_SERVER_URL === undefined)
+    throw new Error('Please provied PROMETHEUS_SERVER_URL in .env')
+
+  const response = await fetch(
+    `${process.env.PROMETHEUS_SERVER_URL}/api/v1/query?query=${metricName}`
+  )
+  const reponseJson = (await response.json()) as {
+    status: string
+    data: { resultType: string; result: any[] }
+  }
```

Comment:
```suggestion
  const reponseJson = z.object({
    status: z.string()
    data: z.object({ resultType: z.string(); result: z.array(z.any()) })
  }).parse(await response.json())
```
This is another place I would use zod